src/predicates/in.jl

```diff
@@ -11,15 +11,15 @@ Base.in(p::Point, g::Geometry) = sideof(p, boundary(g)) == IN
 
 Base.in(p₁::Point, p₂::Point) = p₁ == p₂
 
-function Base.in(p::Point{Dim,T}, s::Segment{Dim,T}) where {Dim,T}
+function Base.in(p::Point{Dim}, s::Segment{Dim}) where {Dim}
   # given collinear points (a, b, p), the point p intersects
   # segment ab if and only if vectors satisfy 0 ≤ ap ⋅ ab ≤ ||ab||²
   a, b = vertices(s)
   ab, ap = b - a, p - a
-  iscollinear(a, b, p) && zero(T) ≤ ab ⋅ ap ≤ ab ⋅ ab
+  iscollinear(a, b, p) && zero(lentype(p)) ≤ ab ⋅ ap ≤ ab ⋅ ab
 end
 
-Base.in(p::Point, r::Ray) = p ∈ Line(r(0), r(1)) && (p - r(0)) ⋅ (r(1) - r(0)) ≥ 0
+Base.in(p::Point, r::Ray) = p ∈ Line(r(0), r(1)) && (p - r(0)) ⋅ (r(1) - r(0)) ≥ zero(lentype(p))^2
```

Comment:
let's refactor this with an auxiliary variable to extract the zero. I believe Julia has the following syntax to evaluate the second expression:

```julia
p ∈ Line(r(0), r(1)) && (x = p - r(0)) ⋅ (r(1) - r(0)); x ≥ zero(x))
```
```